### PR TITLE
feat(data-persistence): Added data persistence on switching between different functions

### PR DESCRIPTION
**Pull request by jatinriverlane**
## Changes
* extension modal submit button now functions as a form
* add test for data persistence

## Features covered
* Keep the data while changing between actions(FKA functions)
* On successful execution of the function form data should be cleared
* It should be a Form with onSubmit and pressing enter as a submission

### CODE REVIEW

File: aqueductcore/frontend/src/components/organisms/ExtensionModal/index.tsx [hunk 1]
```diff
@@ -120,19 +119,43 @@ function ExtensionModal({ isOpen, handleClose, selectedExtension }: ExtensionMod
...
     const { setSelectedFile } = useContext(FileSelectStateContext);
 
     useEffect(() => {
-        setInputParams(selectedAction?.parameters.map(
-            item => (item.dataType === ExtensionParameterDataTypes.EXPERIMENT
-                ? { name: item.name, value: experimentIdentifier }
-                : { name: item.name, value: item.defaultValue ?? "" }
-            ))
-        )
+        if (!isOpen) {
```
Comment (safeamiiir) [inline aqueductcore/frontend/src/components/organisms/ExtensionModal/index.tsx:126]: let's change this useEffect into a function `handleCloseModal ` and call that instead

File: aqueductcore/frontend/src/components/organisms/ExtensionModal/index.tsx [hunk 2]
```diff
@@ -180,53 +210,55 @@ function ExtensionModal({ isOpen, handleClose, selectedExtension }: ExtensionMod
...
-                    </ModalInputsGrid>
-                </ModalMain>
-                <ModalFooter>
-                    {loading ? <CircularProgress size={36} sx={{ mr: 3 }} /> : null}
-                    <RunExtension
-                        variant="contained"
-                        onClick={() => handleExecuteExtension()}
-                        title='run_extension'
-                        disabled={!isExtensionExecutable}
+                <form onSubmit={handleExecuteExtension}>
```
Comment (safeamiiir) [inline aqueductcore/frontend/src/components/organisms/ExtensionModal/index.tsx:213]: Helpful! Thanks

**Review by safeamiiir — APPROVED** [review]
Thanks Jatin! quick changes

File: aqueductcore/frontend/src/components/organisms/ExtensionModal/index.tsx (hunk 1 above)
Comment (jatinriverlane) [reply aqueductcore/frontend/src/components/organisms/ExtensionModal/index.tsx]: Sure, have applied this, thanks.

**Review by samiralavi — APPROVED** [review]
Thanks @jatinriverlane, sorry for late review.

**Pull request merged**